amazon-kinesis-client/src/test/java/software/amazon/kinesis/lifecycle/ConsumerModeUpdateIntegrationTest.java

```diff
@@ -0,0 +1,283 @@
+package software.amazon.kinesis.lifecycle;
+
+import static org.junit.Assert.assertEquals;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+
+import lombok.AllArgsConstructor;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.services.dynamodb.model.ScanRequest;
+import software.amazon.awssdk.services.dynamodb.model.ScanResponse;
+import software.amazon.kinesis.application.ConsumerApplication;
+import software.amazon.kinesis.application.ProducerApplication;
+import software.amazon.kinesis.common.InitialPositionInStreamExtended;
+import software.amazon.kinesis.common.StreamConfig;
+import software.amazon.kinesis.common.StreamIdentifier;
+import software.amazon.kinesis.config.KCLAppConfig;
+import software.amazon.kinesis.config.ReleaseCanaryPollingH2TestConfig;
+import software.amazon.kinesis.config.ReleaseCanarySingleStreamCompatiblePollingH2TestConfig;
+import software.amazon.kinesis.config.ReleaseCanarySingleStreamUpgradePollingH2TestConfig;
+import software.amazon.kinesis.config.multistream.ReleaseCanaryMultiStreamPollingH2TestConfig;
+import software.amazon.kinesis.processor.FormerStreamsLeasesDeletionStrategy;
+import software.amazon.kinesis.processor.MultiStreamTracker;
+import software.amazon.kinesis.processor.StreamTracker;
+import software.amazon.kinesis.utils.RecordValidatorQueue;
+
+@RunWith(Enclosed.class)
+public class ConsumerModeUpdateIntegrationTest {
```

Comment:
Can we add a description for the test?

Reply:
Will address this with follow-up commits.